src/collection.ts

```diff
@@ -235,44 +213,16 @@ export class Collection<T extends TigrisCollectionType> {
 	}
 
 	findMany(
-		filter: SelectorFilter<T> | LogicalFilter<T> | Selector<T>,
+		filter?: SelectorFilter<T> | LogicalFilter<T> | Selector<T>,
 		readFields?: ReadFields,
 		tx?: Session,
 		options?: ReadRequestOptions
-	): Promise<Array<T>> {
-		return new Promise<Array<T>>((resolve, reject) => {
-			if (options === undefined) {
-				options = new ReadRequestOptions();
-			}
-			const result: Array<T> = new Array<T>();
-			this.findManyStream(
-				filter,
-				{
-					onEnd() {
-						resolve(result);
-					},
-					onNext(item: T) {
-						result.push(item);
-					},
-					// eslint-disable-next-line @typescript-eslint/no-unused-vars
-					onError(_error: Error) {
-						reject(_error);
-					},
-				},
-				readFields,
-				tx,
-				options
-			);
-		});
-	}
+	): FindCursor<T> {
+		// find all
+		if (filter === undefined) {
+			filter = { op: SelectorFilterOperator.NONE };
+		}
 
-	findManyStream(
-		filter: SelectorFilter<T> | LogicalFilter<T> | Selector<T>,
-		reader: ReaderCallback<T>,
```

Comment:
Now with the absence of callback API - users are forced to make their caller `async` if they want to consume result in streaming manner. This could trickle a wide refactor in user's code transitively. It will be nice to add support for callback in API shortly as we discussed.

Reply:
We should be looking to provide users the way to stream data with and without forcing them to go in async function.

Reply:
I would very much like us to at this point take inspiration from well liked SDKs such as the MongoDB one. MDB has one of the best user experiences. Once we have grown to be more mature we could devise different ways of doing things. But at this point the MDB SDK maintainers have had far more user feedback received and far more experience building and maintaining the SDK that it is best to follow their lead here.